access_application.go

```diff
@@ -26,33 +26,39 @@ const (
-	CustomDenyURL           string                        `json:"custom_deny_url,omitempty"`
-	HttpOnlyCookieAttribute bool                          `json:"http_only_cookie_attribute,omitempty"`
-	SameSiteCookieAttribute string                        `json:"same_site_cookie_attribute,omitempty"`
-	LogoURL                 string                        `json:"logo_url,omitempty"`
-	SkipInterstitial        bool                          `json:"skip_interstitial,omitempty"`
-	AppLauncherVisible    bool                            `json:"app_launcher_visible,omitempty"`
+	ID                      string                         `json:"id,omitempty"`
+	CreatedAt               *time.Time                     `json:"created_at,omitempty"`
+	UpdatedAt               *time.Time                     `json:"updated_at,omitempty"`
+	AUD                     string                         `json:"aud,omitempty"`
+	Name                    string                         `json:"name"`
+	Domain                  string                         `json:"domain"`
+	Type                    AccessApplicationType          `json:"type,omitempty"`
+	SessionDuration         string                         `json:"session_duration,omitempty"`
+	AutoRedirectToIdentity  bool                           `json:"auto_redirect_to_identity,omitempty"`
+	EnableBindingCookie     bool                           `json:"enable_binding_cookie,omitempty"`
+	AllowedIdps             []string                       `json:"allowed_idps,omitempty"`
+	CorsHeaders             *AccessApplicationCorsHeaders  `json:"cors_headers,omitempty"`
+	CustomDenyMessage       string                         `json:"custom_deny_message,omitempty"`
+	CustomDenyURL           string                         `json:"custom_deny_url,omitempty"`
+	HttpOnlyCookieAttribute bool                           `json:"http_only_cookie_attribute,omitempty"`
+	SameSiteCookieAttribute string                         `json:"same_site_cookie_attribute,omitempty"`
+	LogoURL                 string                         `json:"logo_url,omitempty"`
+	SkipInterstitial        bool                           `json:"skip_interstitial,omitempty"`
+	AppLauncherVisible      bool                           `json:"app_launcher_visible,omitempty"`
+	GatewayRules            []AccessApplicationGatewayRule `json:"gateway_rules,omitempty"`
+	PrivateAddress          string                         `json:"private_address"`
+}
+
+type AccessApplicationGatewayRule struct {
+	ID string `json:"id,omitempty"`
 }
 
 // AccessApplicationCorsHeaders represents the CORS HTTP headers for an Access
 // Application.
 type AccessApplicationCorsHeaders struct {
 	AllowedMethods   []string `json:"allowed_methods,omitempty"`
 	AllowedOrigins   []string `json:"allowed_origins,omitempty"`
-	AllowedHeaders   []string `json:"allowed_headers,omitempty"`
+	AllowedHeaders   []string `json:"gallowed_headers,omitempty"`
```

Comment:
i'm guessing this needs to be

```suggestion
	AllowedHeaders   []string `json:"allowed_headers,omitempty"`
```